test/Graphics/Image.cpp

```diff
@@ -0,0 +1,151 @@
+#include <SFML/Graphics/Image.hpp>
+#include "GraphicsUtil.hpp"
+#include <array>
+
+#include <doctest.h>
+
+TEST_CASE("sf::Image - [graphics]")
+{
+    SUBCASE("Construction")
+    {
+        SUBCASE("Default constructor")
+        {
+            const sf::Image image;
+            CHECK(image.getSize() == sf::Vector2u());
+            CHECK(image.getPixelsPtr() == nullptr);
+        }
+    }
+
+    SUBCASE("Create")
+    {
+        SUBCASE("create(Vector2, Color)")
+        {
+            sf::Image image1;
+            image1.create(sf::Vector2u(10, 10), sf::Color::Red);
+
+            CHECK(image1.getSize() == sf::Vector2u(10, 10));
+            CHECK(image1.getPixelsPtr() != nullptr);
+
+            for (sf::Uint32 i = 0; i < 10; ++i)
+            {
+                for (sf::Uint32 j = 0; j < 10; ++j)
+                {
+                    CHECK(image1.getPixel(sf::Vector2u(i, j)) == sf::Color::Red);
+                }
+            }
+
+            sf::Image image2;
+            image2.create(sf::Vector2u(10, 10));
+            CHECK(image2.getSize() == sf::Vector2u(10, 10));
+            CHECK(image2.getPixelsPtr() != nullptr);
+
+            for (sf::Uint32 i = 0; i < 10; ++i)
+            {
+                for (sf::Uint32 j = 0; j < 10; ++j)
+                {
+                    CHECK(image2.getPixel(sf::Vector2u(i, j)) == sf::Color(0, 0, 0));
+                }
+            }
+        }
+
+        SUBCASE("create(Vector2, Uint8*)")
+        {
+            // 10 x 10, with 4 colour channels array
+            std::array<sf::Uint8, 400> pixels;
+            for (std::size_t i = 0; i < pixels.size(); i += 4)
+            {
+                pixels[i] = 255;     // r
+                pixels[i + 1] = 0;   // g
+                pixels[i + 2] = 0;   // b
+                pixels[i + 3] = 255; // a
+            }
```

Comment:
If we use an array of colors, we can easy initialize all elements with a standard algorithm instead of this loop.